pytorch_lightning/strategies/ddp.py

```diff
@@ -153,17 +161,21 @@ def setup_environment(self) -> None:
     def setup(self, trainer: "pl.Trainer") -> None:
         super().setup(trainer)
         # share ddp pids to all processes
-        self._rank_0_has_called_call_children_scripts = self.broadcast(self._rank_0_has_called_call_children_scripts)
+        self._rank_0_has_called_call_children_scripts = (
+            self.broadcast(self._rank_0_has_called_call_children_scripts) is True
+        )
         if self._should_run_deadlock_detection():
             self._share_information_to_prevent_deadlock()
 
         # move the model to the correct device
         self.model_to_device()
 
+        assert self.model is not None
```

Comment:
Since `self.model` is not used below, let's move this inside `if self.sync_batchnorm`: